plugins/modules/datadog_monitor.py

```diff
@@ -396,7 +415,6 @@ def install_monitor(module):
         "renotify_interval": module.params['renotify_interval'],
         "escalation_message": module.params['escalation_message'],
         "notify_audit": module.boolean(module.params['notify_audit']),
-        "locked": module.boolean(module.params['locked']),
```

Comment:
Is it possible to obtain an API version or something that indicates whether `locked` is accepted or not? That would allow to change the code to work fine while `locked` is still supported, and stop sending it when it's no longer supported.

Reply:
Not that I am aware of it. Datadog will reject any calls to the API with the locked parameter, in the near future. I can't find official date that is announced publicly. 

Though the locked parameter is deprecated in the python api client since [March 2022](https://github.com/DataDog/datadog-api-client-python/pull/888) and the restricted_roles parameter is available since then. [See the datadog owned terraform module](https://github.com/DataDog/terraform-provider-datadog/pull/1392). 

Two weeks ago [datadog did a brownout](https://github.com/DataDog/terraform-provider-datadog/issues/2327#issuecomment-2004961056) and rejected all API calls with the locked parameters.

https://github.com/DataDog/terraform-provider-datadog/pulls?q=is%3Apr+is%3Aclosed+locked+

Reply:
From https://github.com/DataDog/terraform-provider-datadog/issues/2327#issuecomment-2004961056 and the comments below it it seems that only `locked=true` is problematic, but not `locked=false`. If that's true indeed, it would be much less a problem, then users simply have to stop setting `locked` to `true` and instead start using the new `restricted_roles` parameter.

Reply:
Datadog says in their [docs](https://docs.datadoghq.com/monitors/guide/how-to-set-up-rbac-for-monitors/?tab=ui#api) : 

> The locked parameter corresponding to the above mentioned locking mechanism is no longer supported. 

From the support email: 

> Hello,
> We are reaching out to you regarding the deprecation of the "locked" attribute for monitors because you are an admin user of the Safety io Datadog account.
> We noticed your org still have some monitors using the locked attribute and new monitors have been created with this attribute during the last week.
> If you plan to migrate these monitors (see guide [here](https://docs.datadoghq.com/monitors/guide/how-to-set-up-rbac-for-monitors/?tab=ui#api), please let us know as soon as possible by replying to this email.
> If we don't receive any response by April 12, 2024 we will enforce a migration on all the monitors using “locked” to use “restricted_roles” instead. Any tools that leverage our APIs (see [here](https://docs.datadoghq.com/monitors/guide/how-to-set-up-rbac-for-monitors/?tab=ui#api) to check what are the affected endpoints) will start to surface errors and very likely fail when executed if the “locked” is used (e.g. Terraform resources that use the "locked" option).
> Please do not hesitate to ask any questions, we are looking forward to support you with this deprecation.

Reply:
Hmm, how about changing the module as follows then: if `locked=false`, do not put `locked` into `options` in `install_monitor()`.

The main thing I'm worried about is in case:
1. If an existing monitor has `locked=true` set. Will not passing `locked` at all on updating will change `locked`? And if it does, is `changed=true` reported by the module?
2. If a user sets `locked=true` for a monitor which isn't locked right now, is `changed=true` still correctly reported by the module?

The second point is less important, but the first one is.

Reply:
Hey Felix, I do not understand why 'locked' would be still needed if Datadog is removing it from their API. Once they have removed it, you can not create any monitors with the locked parameter and this will cause this module to fail.

Reply:
It's needed for **backwards compatibility**. If you have an existing task that sets `locked=false`, it should keep working. If someone sets `locked=true` and that stops working due to the API no longer supporting it, well, that's a breaking change not caused by this collection. But simply removing this option is a breaking change on our side, and that's not acceptable.

Reply:
Thank you for the explanation that makes perfect sense.